path_role.go

```diff
@@ -0,0 +1,433 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"regexp"
+	"strings"
+)
+
+// Constant to store the current role storage version
+var (
+	currentRoleStorageVersion = 1
+)
+
+// Constants for role specific data
+const (
+	DefaultTTLSeconds = 1800
+	MaxRoleNameLength = 50
+	// Increasing this above this limit might require implementing
+	// client-side paging in the filterGroupMembership API
+	MaxOCIDsPerRole = 100
+)
+
+func pathRole(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "role/" + framework.GenericNameRegex("role"),
+		Fields: map[string]*framework.FieldSchema{
+			"role": {
+				Type:        framework.TypeString,
+				Description: "Name of the role.",
+			},
+			"description": {
+				Type:        framework.TypeString,
+				Description: `A description of the role.`,
+			},
+			"add_ocid_list": {
+				Type:        framework.TypeCommaStringSlice,
+				Description: `A comma separated list of OCIDs to add.`,
+			},
+			"remove_ocid_list": {
+				Type:        framework.TypeCommaStringSlice,
+				Description: `A comma separated list of OCIDs to remove. Is applicable only for the UPDATE operation.`,
+			},
+			"ttl": {
+				Type:        framework.TypeInt,
+				Default:     0,
+				Description: `Duration in seconds after which the issued token should expire. Defaults to 1800 seconds.`,
+			},
+			"add_policy_list": {
+				Type:        framework.TypeCommaStringSlice,
+				Default:     "default",
+				Description: "A list of Policies to be set on tokens issued using this role.",
+			},
+			"remove_policy_list": {
+				Type:        framework.TypeCommaStringSlice,
+				Default:     "default",
+				Description: "A list of Policies to be remove from the role. Is applicable only for the UPDATE operation.",
+			},
+		},
+
+		ExistenceCheck: b.pathRoleExistenceCheck,
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.CreateOperation: b.pathRoleCreate,
+			logical.UpdateOperation: b.pathRoleUpdate,
+			logical.ReadOperation:   b.pathRoleRead,
+			logical.DeleteOperation: b.pathRoleDelete,
+		},
+
+		HelpSynopsis:    pathRoleSyn,
+		HelpDescription: pathRoleDesc,
+	}
```

Comment:
Save the path as a variable first and then use `tokenutil.AddTokenFields(p.Fields)` which will add a number of common role parameters. [Example](https://github.com/hashicorp/vault/blob/c48159ea3afd1cf5fee9a08b9b3475a615700fd4/builtin/credential/aws/path_role.go#L189)

I made a related comment about the RoleEntry structure further down.

Reply:
Done, thanks for the info.